core/src/main/java/io/substrait/relation/VirtualTableScan.java

```diff
@@ -24,13 +24,16 @@ public abstract class VirtualTableScan extends AbstractReadRel {
   @Value.Check
   protected void check() {
     var names = getInitialSchema().names();
+
+    assert names.size()
+        == NamedFieldCountingTypeVisitor.countNames(this.getInitialSchema().struct());
```

Comment:
It may make more sense to move this check into the NamedStruct itself in the long-term. Avoiding this for now because there's some stuff I'd like to verify around names going to and from Calcite and Substrait.